startup_scripts/20_custom_fields.py

```diff
@@ -56,13 +56,14 @@ def get_class_for_class_path(class_path):
 
         custom_field.save()
 
-        for choice_details in cf_details.get('choices', []):
-          choice = CustomFieldChoice.objects.create(
+        for idx, choice_details in enumerate(cf_details.get('choices', [])):
+          choice, created = CustomFieldChoice.objects.get_or_create(
             field=custom_field,
-            value=choice_details['value'])
-
-          if choice_details.get('weight', 0):
-            choice.weight = choice_details['weight']
-            choice.save()
+            value=choice_details['value'],
+          )
+          # Add weight after initial creation to fix a bug if you use the same 'value'
+          # for multiple custom fields.
+          choice.weight = choice_details.get('weight', idx * 10)
```

Comment:
Why do you multiply the index by 10?

Reply:
I think you can just add the `weight` as part of [the `defaults` hash in the arguments to `get_or_create`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.get_or_create):

```python
choice = CustomFieldChoice.objects.get_or_create(
  field=custom_field,
  value=choice_details['value'],
  defaults={'weight': idx}
)
```

Could you please check this?

Reply:
In my setup i found out that if you have the same choice value on 2 different fields it will cause a exception. I will try to make a small example so you can test it yourself. By creating the object before i set the weight value, it solved the issue.

Reply:
I understand that. But the documentation says:

> Any keyword arguments passed to get_or_create() — except an optional one called defaults — will be used in a get() call.